pyvista/plotting/renderer.py

```diff
@@ -880,21 +881,18 @@ def add_actor(
         if isinstance(actor, _vtk.vtkMapper):
             actor = Actor(mapper=actor, name=name)
 
-        if isinstance(actor, Actor) and name:
-            # WARNING: this will override the name if already set on Actor
-            actor.name = name
-
         if name is None:
-            # Fallback for non-wrapped actors
-            # e.g., vtkScalarBarActor
-            name = actor.name if isinstance(actor, Actor) else actor.GetAddressAsString('')
-
+            name = (
+                actor.name
+                if (hasattr(actor, 'name') and actor.name)
+                else actor.GetAddressAsString('')
```

Comment:
Can change this to better match the naming used by `Actor` (and related) classes
```suggestion
                else f'{type(actor).__name__}({actor.GetAddressAsString('')})'
```

Naming used by actors:

https://github.com/pyvista/pyvista/blob/d1c369dd79b6ab4d92a10bdbd123b426aac0fa0c/pyvista/core/utilities/misc.py#L362-L367

Reply:
+1